seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProvider.java

```diff
@@ -292,15 +325,35 @@ public HttpResponse doPost(
     /**
      * Send a post request with request headers , request parameters and request body
      *
-     * @param url request address
      * @param headers request header map
      * @param params request parameter map
      * @param body request body
      * @return http response result
      * @throws Exception information
      */
     public HttpResponse doPost(
-            String url, Map<String, String> headers, Map<String, String> params, String body)
+            URI uri,
+            Map<String, String> headers,
+            Map<String, Object> params,
+            Map<String, Object> body,
+            boolean keepParamsAsForm)
```

Comment:
doPost should not care `keepParamsAsForm` if we can handle old logic in https://github.com/apache/seatunnel/pull/8434/files#diff-84c2e7b09f0d771042057899d626749a07479adaa72a7db60d17ed0ec3592448R130